test/src/kbasesearchengine/test/events/handler/WorkspaceEventHandlerTest.java

```diff
@@ -696,4 +703,528 @@ private void failLoadWSGetWSInfoException(final Exception toThrow, final Excepti
             TestCommon.assertExceptionCorrect(got, expected);
         }
     }
+
+    @Test
+    public void updateEventAccessGroupEvent() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                "WS",
+                           Instant.ofEpochMilli(20000),
+                           StatusEventType.PUBLISH_ACCESS_GROUP)
+                           .withNullableAccessGroupID(1)
+                           .build();
+
+        when(wscli.administer(argThat(new AdminGetWSInfoAnswerMatcher(1))))
+                .thenReturn(new UObject(wsTuple(1, "wsname", "username", "date", 7, "r", "n",
+                        "unlocked", Collections.emptyMap())));
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        // update event
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        // access group events are not updated, so we expect to get back the original event
+        Assert.assertEquals("expected the same event object", event, updatedEvent);
+    }
+
+    @Test
+    public void updateEventWorkspacePermanentlyDeleted() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.RENAME_ALL_VERSIONS)
+                .withNullableNewName("newName")
+                .withNullableAccessGroupID(1)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        when(wscli.administer(argThat(new AdminGetWSInfoAnswerMatcher(1))))
+                .thenThrow(new JsonClientException("No workspace with id 1 exists!"));
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        // update event
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        // since workspace does not exist, we expect to get back an updated event
+        Assert.assertNotSame("expected different event object", event, updatedEvent);
+        Assert.assertEquals(updatedEvent.getEventType(), StatusEventType.DELETE_ALL_VERSIONS);
```

Comment:
Why not just build the event you expect and assertThat("event incorrect", event1, is(event2))?

Reply:
Done.